content/qemu/index.md

```diff
@@ -3,40 +3,97 @@ title = 'QEMU'
 icon = 'clanbomber'
 +++
 
-<a href="https://www.qemu.org/">QEMU</a> is an emulator for
-various CPUs. It works on Linux, Windows, FreeBSD and Mac OS X. It's possible
-to run grml with QEMU. Notice that it's much slower than running it native
-due to the emulation mode.
+[QEMU](https://www.qemu.org/) is an emulator for various CPUs.
+It works on Linux, Windows, FreeBSD and Mac OS X.
+It's possible to run grml with QEMU.
+Note that it is slower than running it native due to the emulation mode.
 
 ## Using QEMU with acceleration
 
-grml >=0.5 provides the <a href="https://www.qemu.org/">accelerator module</a> for QEMU namend 'kqemu'. Just run the following commands to use it:
+On modern Linux distributions, including Grml, the kernel comes with an acceleration module, called `KVM`.
+To use this, load the `kvm.ko` kernel module, and give yourself rights:
 
 <pre class="rahmen">
-apt-get update ; apt-get install qemu # install qemu
-modprobe kqemu                        # load the kernel module
-mknod /dev/kqemu c 250 0              # create the device
-chmod 666 /dev/kqemu /dev/net/tun     # adjust permissions
+apt update && apt install qemu-system        # install qemu
+modprobe kvm                                 # load the kernel module
+adduser $USER kvm                            # add your user to kvm group
+# now login again or try: newgrp kvm
 </pre>
 
-To see if kqemu is enabled and working correctly, use the QEMU monitor command (press ctrl-alt-2): 'info kqemu'
+To see if kvm is enabled and working correctly, use the QEMU monitor command (press ctrl-alt-2): 'info kvm'
```

Comment:
Hmm, I run `-cpu host -enable-kvm` without being in the `kvm` group. Is this still necessary? How to test that?

`info kvm` looks like this on my system:
```
(qemu) info kvm
kvm support: enabled
```

Reply:
Hmm, what are the permissions on /dev/kvm on your machine then?

Reply:
On a random machine I get this:
```
% getfacl /dev/kvm
getfacl: Removing leading '/' from absolute path names
# file: dev/kvm
# owner: root
# group: kvm
user::rw-
user:ch:rw-
group::rw-
mask::rw-
other::---
```

I *guess* systemd/logind adds an ACL for my user as I'm logged in locally. I think this is not the case if logged in remotely.

Reply:
yes, I have the same permissions. I am still not sure what that means for me...

Reply:
well i think that means if you use `ssh` and are not logged in locally, you must be in the `kvm` group.

Reply:
reworded the text there to reflect that it works as is if logged in locally

Reply:
I just logged in via SSH to localhost. This works `qemu-system-x86_64 -M q35 -m 2048 -cpu host -enable-kvm -nographic -cdrom /srv/isos/grml-full-2024.12-amd64.iso` without being in the `kvm` group. I need to look into that.

Reply:
> [...] not logged in locally

🙈 I will test without being logged in locally.